### PR TITLE
fix(core): get data nullifier logic

### PR DESCRIPTION
**Pull request by vimwitch**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint --check`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/src/UserState.ts [hunk 1]
```diff
@@ -237,6 +237,23 @@ export default class UserState {
...
+            allNullifiers.push(
+                ...[0, this.sync.settings.numEpochKeyNoncePerEpoch].map((v) =>
+                    genEpochKey(this.id.secret, attesterId, x, v).toString()
+                )
+            )
+        }
+        const foundNullifiers = await this.sync._db.findMany('Nullifier', {
+            where: {
+                attesterId,
+                nullifier: allNullifiers,
```
Comment (vivianjeng) [inline packages/core/src/UserState.ts:251]: Is it an `OR` logic of `allNullifiers`?

File: packages/core/src/UserState.ts [hunk 2]
```diff
@@ -237,6 +237,23 @@ export default class UserState {
...
+            },
+        })
+        const sortedNullifiers = foundNullifiers.sort((a, b) =>
+            a.epoch > b.epoch ? 1 : -1
+        )
```
Comment (vivianjeng) [inline packages/core/src/UserState.ts:256]: can it order by
```ts
await this.sync._db.findMany('Nullifier', {
   where: {
   ...
   },
   orderBy: {
      epoch: 'asc'
   }
}
```
?

File: packages/core/src/UserState.ts (hunk 1 above)
Comment (vimwitch) [reply packages/core/src/UserState.ts:251]: yes, it selects documents where `nullifier` is a value in `allNullifiers`

**Review by vivianjeng — APPROVED** [review]
LGTM

**Pull request merged**